android/src/test/java/io/radar/react/RNRadarModuleTest.java

```diff
@@ -0,0 +1,249 @@
+import com.facebook.react.bridge.ReadableMap;
+import com.facebook.react.bridge.WritableArray;
+import com.facebook.react.bridge.WritableMap;
+import io.radar.sdk.Radar;
+import io.radar.sdk.Radar.RadarCallback;
+import io.radar.sdk.Radar.RadarPlacesProvider;
+import io.radar.sdk.Radar.RadarStatus;
+import io.radar.sdk.RadarTrackingOptions;
+import io.radar.sdk.model.RadarEvent;
+import io.radar.sdk.model.RadarGeofence;
+import io.radar.sdk.model.RadarUser;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.invocation.InvocationOnMock;
+import org.mockito.stubbing.Answer;
+import org.powermock.api.mockito.PowerMockito;
+import org.powermock.core.classloader.annotations.PrepareForTest;
+import org.powermock.modules.junit4.PowerMockRunner;
+
+import static org.junit.Assert.assertEquals;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyBoolean;
+import static org.mockito.ArgumentMatchers.anyInt;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+import static org.powermock.api.mockito.PowerMockito.doAnswer;
+import static org.powermock.api.mockito.PowerMockito.verifyStatic;
+import static org.powermock.api.mockito.PowerMockito.when;
+
+@RunWith(PowerMockRunner.class)
+@PrepareForTest({Radar.class, RNRadarUtils.class, ActivityCompat.class,
+    ContextCompat.class, Arguments.class})
+public class RNRadarModuleTest {
+
+  @Mock
+  private ReactApplicationContext context;
+
+  private RNRadarModule module;
+
+  @Before
+  public void setup() {
+    PowerMockito.mockStatic(Radar.class);
+    PowerMockito.mockStatic(RNRadarUtils.class);
+    PowerMockito.mockStatic(ActivityCompat.class);
+    PowerMockito.mockStatic(ContextCompat.class);
+    PowerMockito.mockStatic(Arguments.class);
+    when(Arguments.createMap()).then(new Answer<WritableMap>() {
+      @Override
+      public WritableMap answer(InvocationOnMock invocation) {
+        return new JavaOnlyMap();
+      }
+    });
+    when(Arguments.createArray()).then(new Answer<WritableArray>() {
+      @Override
+      public WritableArray answer(InvocationOnMock invocation) {
+        return new JavaOnlyArray();
+      }
+    });
+
+    module = new RNRadarModule(context);
+  }
+
+  @Test
+  public void getName() {
+    assertEquals("RNRadar", module.getName());
+  }
+
+  @Test
+  public void setUserId() {
+    String userId = "userId123";
+
+    module.setUserId(userId);
+
+    verifyStatic(Radar.class);
+    Radar.setUserId(userId);
+  }
+
+  @Test
+  public void setDescription() {
+    String description = "some user description";
+
+    module.setDescription(description);
+
+    verifyStatic(Radar.class);
+    Radar.setDescription(description);
+  }
+
+  @Test
+  public void setPlacesProvider() {
+    String providerStr = "facebook2";
```

Comment:
facebook2?

Reply:
oops must've slipped, doesn't matter anyways since this is just a random string